okhttp/src/main/java/io/grpc/okhttp/OkHttpTlsUpgrader.java

```diff
@@ -74,6 +78,14 @@ public static SSLSocket upgrade(SSLSocketFactory sslSocketFactory,
           OkHttpProtocolNegotiator.get().negotiate(sslSocket, host, TLS_PROTOCOLS);
       Preconditions.checkState(HTTP2_PROTOCOL_NAME.equals(negotiatedProtocol),
           "Only \"h2\" is supported, but negotiated protocol is %s", negotiatedProtocol);
+
+      SSLSession session = sslSocket.getSession();
+      if (session == null) {
+        throw new SSLException("Cannot get the ssl session");
+      }
+      if (!OkHostnameVerifier.INSTANCE.verify(host, session)) {
```

Comment:
Good catch, actually staring the handshake should also be independent of `supportsTlsExtensions`

Fixed.